SessionUploadExample/Program.cs

```diff
@@ -187,30 +192,44 @@ private static void Main(string[] args)
             string serverDns,
             string directory,
             Guid folderId,
-            bool continueOnError)
+            bool continueOnError,
+            string compatibility)
         {
             Dictionary<string, SessionUpload> uploads = new Dictionary<string, SessionUpload>();
 
             // collect all XML files in the folder
             string[] xmlFilePaths = Directory.EnumerateFiles(directory, "*.xml", SearchOption.AllDirectories).ToArray();
 
-            // errors encountered while deserializing XML files as PanoptoSessions
+            // errors encountered while deserializing XML files
             Dictionary<string, Exception> deserializationExceptions = new Dictionary<string,Exception>();
 
-            // a collection of all the PanoptoSession XML files found and the list of files that session references
+            // a collection of all the UCS XML files found and the list of files that session references
             Dictionary<string, string[]> referencedFilesByManifest = new Dictionary<string, string[]>();
 
             // try to parse each as a Panopto session
             foreach (string xmlFilePath in xmlFilePaths)
             {
-                PanoptoSession panoptoSession = null;
-                
                 using (FileStream xmlFileStream = File.Open(xmlFilePath, FileMode.Open))
                 {
                     try
                     {
-                        XmlSerializer deserializer = new XmlSerializer(typeof(PanoptoSession));
-                        panoptoSession = (PanoptoSession)deserializer.Deserialize(xmlFileStream);
+                        // collect filenames in the session to see if they exist
+                        IEnumerable<string> referencedFiles;
+                        if (compatibility == "v1")
```

Comment:
Maybe a simpler approach would be to use XmlSerializer.CanDeserialize to try both V2 and V1 (in that order) then use the one that works? That way no compatibility flag is needed

Reply:
Updated. I ordered it so it tests for not being v1, so anything that isn't either v1 or v2 (i.e. is invalid) will get an error message from the v2 parser instead of the v1 parser.